### PR TITLE
Error dialog when a evaluator plugin fails to load

### PR DESCRIPTION
**Pull request by scottdraves**
close Issue #1180

### CODE REVIEW

File: core/src/main/web/app/helpers/evaluatepluginmanager.js [hunk 1]
```diff
@@ -129,9 +129,18 @@
                   .then(function(factory) {
                     if (factory !== undefined && factory.create !== undefined) {
                       return factory.create(evaluatorSettings).then(function(ev) { deferred.resolve(ev); });
-                    } else
-                      deferred.reject("no factory for evaluator plugin");
+                    } else {
+		      // or should someone pass in a scope to use, like with a method on the evaluatorLoadQueue?
+                      $rootScope.pluginId = name;
```
Comment (buob) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: did `$rootScope.$new({ pluginId: name })` etc. etc. not work?

Comment (scottdraves) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: i tried a few things like that but did not fly.  would be happy to see it though.

File: core/src/main/web/app/helpers/evaluatepluginmanager.js [hunk 2]
```diff
@@ -129,9 +129,18 @@
...
+		      var modal = $modal.open({backdrop: true,
+					       backdropClick: true,
+					       template: JST['helpers/plugin-load-error']()});
```
Comment (buob) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: You could also pass in a variable directly to the template invocation (especially considering it's not going to change at all after it's displayed), trying to find an example...

Comment (buob) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: here's a [js file](https://github.com/twosigma/beaker-notebook/blob/2b354551cca27d8c42663ae0f2234eecb77abf12/core/src/main/web/plugin/menu/file_filesys.js#L39) passing in a variable to [template](https://github.com/twosigma/beaker-notebook/blob/2b354551cca27d8c42663ae0f2234eecb77abf12/core/src/main/web/app/template/opennotebook.jst.html#L33)

Comment (buob) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: Might have to use [this syntax](https://github.com/twosigma/beaker-notebook/blob/2b354551cca27d8c42663ae0f2234eecb77abf12/core/src/main/web/app/template/opennotebook.jst.html#L40) (like extension) -- I was having trouble using the other in our environment

File: core/src/main/web/app/helpers/evaluatepluginmanager.js [hunk 3]
```diff
@@ -129,9 +129,18 @@
                   .then(function(factory) {
                     if (factory !== undefined && factory.create !== undefined) {
                       return factory.create(evaluatorSettings).then(function(ev) { deferred.resolve(ev); });
-                    } else
-                      deferred.reject("no factory for evaluator plugin");
+                    } else {
+		      // or should someone pass in a scope to use, like with a method on the evaluatorLoadQueue?
+                      $rootScope.pluginId = name;
+		      var modal = $modal.open({backdrop: true,
```
Comment (buob) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: you probably don't need this `modal` var stored, do you?

File: core/src/main/web/app/helpers/evaluatepluginmanager.js (hunk 2 above)
Comment (scottdraves) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: xlnt thank you, indeed the ${} syntax is required

File: core/src/main/web/app/helpers/evaluatepluginmanager.js (hunk 3 above)
Comment (scottdraves) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: right, done

File: core/src/main/web/app/helpers/evaluatepluginmanager.js [hunk 4]
```diff
@@ -19,7 +19,7 @@
 (function() {
   'use strict';
   var module = angular.module('bk.evaluatePluginManager', ['bk.utils']);
-  module.factory('bkEvaluatePluginManager', function(bkUtils) {
+  module.factory('bkEvaluatePluginManager', function(bkUtils, $rootScope, $modal) {
```
Comment (buob) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: You probably don't need this dep on `$rootScope` anymore, right?

**Comment by buob** [comment]
Looks good other than small comment :+1:

File: core/src/main/web/app/helpers/evaluatepluginmanager.js (hunk 4 above)
Comment (scottdraves) [inline core/src/main/web/app/helpers/evaluatepluginmanager.js]: thanks, removed

**Pull request merged**